### PR TITLE
Completions in lowercase.

### PR DESCRIPTION
**Pull request by j-bennet**
## Description

Pgcli can do `upper`, `lower` and `auto` casing for completions; ported the same functionality to `mycli`.

Issue: https://github.com/dbcli/mycli/issues/128

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [X] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by mikeerickson** [comment]
+1 please merge :)

**Comment by tsroten** [comment]
@j-bennet This has been on my wishlist for awhile. Thanks for doing this!!

File: mycli/main.py [hunk 1]
```diff
@@ -141,14 +141,16 @@ def __init__(self, sqlexecute=None, prompt=None,
         self.prompt_format = prompt or prompt_cnf or c['main']['prompt'] or \
                              self.default_prompt
         self.prompt_continuation_format = c['main']['prompt_continuation']
+        keyword_casing = c['main'].get('keyword_casing', 'upper')
```
Comment (amjith) [inline mycli/main.py]: Should this default to 'auto'?

File: mycli/myclirc [hunk 2]
```diff
@@ -77,6 +77,9 @@ login_path_as_host = False
 # and using normal tabular format otherwise. (This applies to statements terminated by ; or \G.)
 auto_vertical_output = False
 
+# keyword casing preference. Possible values "lower", "upper", "auto"
+keyword_casing = upper
```
Comment (amjith) [inline mycli/myclirc]: Same here. I think 'auto' might be preferable.

File: mycli/sqlcompleter.py [hunk 3]
```diff
@@ -59,6 +59,9 @@ def __init__(self, smart_completion=True, supported_formats=()):
 
         self.special_commands = []
         self.table_formats = supported_formats
+        if keyword_casing not in ('upper', 'lower', 'auto'):
+            keyword_casing = 'upper'
```
Comment (amjith) [inline mycli/sqlcompleter.py]: 'auto'

File: mycli/myclirc (hunk 2 above)
Comment (j-bennet) [inline mycli/myclirc]: I was thinking more along the lines of preserving current mycli behavior in defaults, but it's true, auto sounds like a better default. I'll change that.

**Comment by j-bennet** [comment]
Ideally, I'd rework the unit tests also, to parametrize the completer with lower case and upper case, but I don't have time for something that massive right now. It will have to be a TODO for later.

**Comment by tsroten** [comment]
@j-bennet First of all, this is awesome!

If I set my keyword casing to "lower", it's getting changed to "auto". I think this happens in `mycli.refresh_completions()` where it has the `dict` of arguments -- we need to add `keyword_casing` there.

**Comment by j-bennet** [comment]
@tsroten Good catch! That should be fixed now (losing the casing setting on refresh).

**Pull request merged**